CMakeLists.txt

```diff
@@ -64,14 +65,49 @@ endif()
 # Dependencies
 # =============================================================================
 
+find_package(MPI)
+if (MPI_FOUND)
+    # With MPI we prefer HDF5 parallel
+    set(HDF5_PREFER_PARALLEL "ON")
+endif()
+
+find_package(HDF5)
```

Comment:
That's an issue for HighFive. I don't see it in the file used to generate the `find_package` magic: https://github.com/BlueBrain/HighFive/blob/master/CMake/HighFiveConfig.cmake.in

Reply:
Spoke too fast. Here it is: https://github.com/BlueBrain/HighFive/blob/master/CMake/HighFiveTargetDeps.cmake#L24

Reply:
Yes, but we needed to use pure HDF5 because at some point we would need compound datatypes (not supported in HighFive).
Although I guess we could find_package(HighFive) and use still pure HDF5.

Reply:
In that case, I think it's better to be explicit?

Reply:
> at some point we would need compound datatypes (not supported in HighFive).

Are there plans to add compound datatypes in `HighFive`?  If we're not using compound datatypes yet here, it would be preferable to only use `HighFive` implementation.

> In that case, I think it's better to be explicit?
Explicit with an explanation, and pointers to when it needs to be removed (ie: when compound dtypes hit HighFive.)

Reply:
> Are there plans to add compound datatypes in HighFive? If we're not using compound datatypes yet here, it would be preferable to only use HighFive implementation.

Adding compound data type enquires quite some efforts (there is[ initial PR](https://github.com/BlueBrain/HighFive/pull/78) but that's waiting for ~2.5 years). With recent push from Fernando/Tristan/Nico in HighFive, we are definitely checking what is missing in HighFive (all parallel aspects + compound dtype) but I would target  that as separate PR.

Reply:
Ok, but since compound datatypes aren't used in this PR, we can just use vanilla `HighFive`, right?